backends/dotnet/src/bootstrap/dllmain.cpp

```diff
@@ -0,0 +1,136 @@
+    {
+        Log().Get() << "DLL_THREAD_ATTACH";
+        break;
+    }
+    case DLL_THREAD_DETACH:
+    {
+        Log().Get() << "DLL_THREAD_DETACH";
+        break; 
+    }
+    case DLL_PROCESS_DETACH:
+    {
+        Log().Get() << "DLL_PROCESS_DETACH";
+        break;
+    }
+    }
+    return TRUE;
+}
+
+
+bool initWorkerDllAbsolutePath(HMODULE hCurrentDll, wchar_t* buf, size_t len, const wchar_t* workerDllName) {
+    Log().Get() << "getting absolute path to worker dll...";
+    int ret = GetModuleFileNameW(hCurrentDll, buf, (DWORD)len);
+    if (!ret) {
+        Log().LogLastError();
+        Log().Get() << "Error: GetModuleFileNameW failed with " << ret;
+        return false;
+    }
+    Log().Get() << "module path is " << buf;
+    PathRemoveFileSpecW(buf);
+    Log().Get() << "module folder is " << buf;
+    wcscat_s(buf, len, L"\\");
+    wcscat_s(buf, len, workerDllName);
+    Log().Get() << "worker dll path is " << buf;
+
+    return true;
+}
+
+extern "C" __declspec(dllexport) 
+int LoadWorkerDll() {
+    Log().Get() << "LoadWorkerDll() called";
+
+    // TODO CorBindToRuntimeEx for .NET < 4.0
+    ICLRMetaHost* metaHost = nullptr;
+    if (CLRCreateInstance(CLSID_CLRMetaHost, IID_ICLRMetaHost, (LPVOID*)&metaHost) == S_OK) {
+        // TODO EnumerateLoadedRuntimes ?
+        const wchar_t* runtimeVersion = L"v4.0.30319";
+        ICLRRuntimeInfo* runtimeInfo = nullptr;
+        if (metaHost->GetRuntime(runtimeVersion, IID_ICLRRuntimeInfo, (LPVOID*)&runtimeInfo) == S_OK) {
+            ICLRRuntimeHost* runtimeHost = nullptr;
+            if (runtimeInfo->GetInterface(CLSID_CLRRuntimeHost, IID_ICLRRuntimeHost, (LPVOID*)&runtimeHost) == S_OK) {
+                Log().Get() << "runtimeHost->Start()...";
+
+                runtimeHost->Start();
+
+                Log().Get() << "load worker dll and start server - " << workerDllPath;
+                DWORD pReturnValue;
+                int ret = runtimeHost->ExecuteInDefaultAppDomain(workerDllPath, L"InjectedWorker.Main", L"StartServer", L"", &pReturnValue);
+                if (ret == S_OK) {
+                    Log().Get() << "worker dll: server is finished";
+                }
+                else {
+                    Log().LogLastError();
+                    Log().Get() << "ExecuteInDefaultAppDomain failed with return value " << ret;
+                }
+
+                //OPTIONAL: You can keep the CLR Opened depending on your needs
+                runtimeHost->Release();
+            }
+            else {
+                Log().LogLastError();
+                Log().Get() << "GetInterface() failed";
+            }
+            runtimeInfo->Release();
+        }
+        else {
+            Log().LogLastError();
+            Log().Get() << "GetRuntime() failed";
+        }
+        metaHost->Release();
+    }
+    else {
+        Log().LogLastError();
+        Log().Get() << "bootstrap: CLRCreateInstance() failed";
+    }
+
+    return 0;
+}
+
+
+
+
+
+/*
+
```

Comment:
Maybe move to the header of module and mention that "this code is inspired by these materials:" or something like that.